lib/call.ts

```diff
@@ -1,20 +1,42 @@
-import type { Operation } from "./types.ts";
+import type { Instruction, Operation } from "./types.ts";
 import { action } from "./instructions.ts";
 import { pause } from "./pause.ts";
 
+function isFunc(f: unknown): boolean {
+  return typeof f === "function";
+}
+function isPromise<T>(p: unknown): p is Promise<T> {
+  if (!p) return false;
+  return isFunc((p as Promise<T>).then);
+}
+// iterator must implement both `.next` and `.throw`
+// built-in iterators are not considered iterators to `call()`
+function isInstructionIterator<T>(it: unknown): it is Iterator<Instruction, T> {
+  if (!it) return false;
+  return isFunc((it as Iterator<Instruction, T>).next) &&
+    isFunc((it as Iterator<Instruction, T>).throw);
+}
+function isIterable<T>(it: unknown): it is Iterable<T> {
+  if (!it) return false;
+  return typeof (it as Iterable<T>)[Symbol.iterator] === "function";
+}
+
 /**
  * Pause the current operation, and run a promises, async functions, or
  * operations within a new scope. The calling operation will resumed (or errored)
```

Comment:
Looks like there's a slight grammatical error here. Might as well fix this now.

```suggestion
 * Pause the current operation, and run a promise, async function, or
 * operation within a new scope. The calling operation will resumed (or errored)
```